vshard/storage/init.lua

```diff
@@ -3180,6 +3180,41 @@ local function storage_ref(rid, timeout)
     return bucket_count()
 end
 
+--
+-- Lookup for absent active buckets.
+--
+-- @param bucket_ids List of bucket identifiers.
+-- @return A dictionary with a list of bucket identifiers which are
+--         not present on the current instance.
+--
+local function storage_moved_buckets(bucket_ids)
+    local allstatus = consts.BUCKET
+    local moved_buckets = {}
+    for _, bucket_id in pairs(bucket_ids) do
+        local bucket = box.space._bucket:get{bucket_id}
+        local is_moved
+        if not bucket then
+            is_moved = true
+        else
+            local status = bucket.status
+            -- All the buckets which can turn into ACTIVE directly or are
+            -- already ACTIVE.
+            is_moved = status ~= allstatus.ACTIVE and
```

Comment:
My idea was that I wanted to explicitly mention potentially active buckets, and all the rest are by default not suitable. So if we would add a new status, it would be treated as moved. But you are right, I've overengineered it a bit. Done as you proposed.